### PR TITLE
Removes ass jam

### PR DESCRIPTION
**Pull request by pali6**
[removal] [input wanted]

## About the PR 
Ass jam suffers from unclear purpose and every month as it rolls around a ton of usually valid overwhelmingly negative feedback appears about it. This PR removes ass jam. Please discuss in the forum thread.

Note: this PR lacks the ass jam label on purpose because that label forces the changelog entry into the ass jam changelog which this PR removes so there would be no point in putting it there.

## Why's this needed? 
Concerns raised by the community:

- Ass jam makes the code ugly and harder to develop when it's not in a separate branch.
- Ass jam lacks a clear purpose, it's a mix of admin shenanigans, powergaming PRs, testing PRs, silly PRs, broken PRs and shitposts.
- It is indistinguishable from a regular round.
- The ass day classic mode is too murderboney.
- Admins don't run events on ass jam and don't give a shit about the round.
- Features get reverted randomly. Moreover this isn't properly noted in the changelog so no one knows what changes are in-game and what aren't.
- No one wants to develop for ass jam because you're only developing a feature to be available for one day instead of for the other 29-30 days of a month. And the lack of clarity about the purpose of ass jam doesn't help. Most projects are either good enough to be enabled permanently or bad enough to not be enabled at all, the margin for ass-jam-ness is small.
- Most ass jam features are minor and not noticed anyway.
- It just isn't fun.
- The wider variety of maps doesn't appeal to everyone.

## Changelog
```
(u)pali
(*)Ass jam has been removed
```

### CODE REVIEW

File: code/mob/living/carbon/human/procs/update_icon.dm [hunk 1]
```diff
@@ -1218,40 +1218,12 @@ var/list/update_body_limbs = list("r_arm" = "stump_arm_right", "l_arm" = "stump_
...
-#endif
-
 	if (src.bioHolder)
 		src.bioHolder.OnMobDraw()
 	//Also forcing the updates since the overlays may have been modified on the images
 	src.UpdateOverlays(src.body_standing, "body", 1, 1)
 	src.UpdateOverlays(src.hands_standing, "hands", 1, 1)
 
-#if ASS_JAM //Oh neat apparently this has to do with cool maptext for your health, very neat. plz comment cool things like this so I know what all is on assjam!
-/mob/living/carbon/human/UpdateDamage()
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/update_icon.dm:1235]: might be better if this is just commented out, for future use, it's actually quite neat?

**Review by ZeWaka — COMMENTED** [review]
Code looks correct. Doesn't address maps nor icons of stuff removed, ex. sonicbee.
Does not remove ass jam stuff from github action pipelines.

**Comment by pali6** [comment]
I'd say that the amount of icons left over is minimal and most of the stuff is still available to be spawned by admin intervention (like sonic bees). Removing it all manually would be a pain in the ass, possibly by reverting the relevant merge commits and using the dmi merge driver might make it easier but even that's not gonna be clean as a bunch of ass jam code got changed after the PR merge (e.g. throwing refactor had to move timestop stuff).

File: code/mob/living/carbon/human/procs/update_icon.dm (hunk 1 above)
Comment (pali6) [reply code/mob/living/carbon/human/procs/update_icon.dm:1235]: The exact same code is used right below the removed part (in the `#else` block) for the murderbox training dummy, I think having an additional commented out copy isn't necessary.

**Comment by pali6** [comment]
GitHub action pipelines should now be ass jam free but please doublecheck if beepsky and automaton setups are correct.

**Comment by Gannets** [comment]
Yeah, it's probably time

**Comment by RSG250** [comment]
i would like an admin button created for the bush shaking thing at least, for the option when yall feel like it

**Review by mordent-goonstation — APPROVED** [review]

**Comment by pali6** [comment]
Warc requested that the code is kept so I'll modify the PR to only remove the trigger for turning on ass jam.

**Review by Tarmunora — CHANGES_REQUESTED** [review]
Disabling assjam but leaving the code, from a code perspective, is the worst possible outcome. This would leave a (small) mess of unmaintained code sitting around becoming ever more unable to be turned on the more things change around it. Either keep it enabled or purge the code from master, but don't do it by halves.

**Comment by pali6** [comment]
So should I revert back to this: https://github.com/goonstation/goonstation/pull/2669/files/f2a3da6009711025fbf299e8a30c806cc210fe19 ?

**Review by Tarmunora — COMMENTED** [review]
The current state of this PR appears to just remove the ass-changelog and ass-related checks/github-things without changing the game?
Confirm if this is intentional.

**Comment by pali6** [comment]
ah wait heck I forgot to revert the last commit, my bad

**Comment by pali6** [comment]
Should be good now. If the PR gets merged it can always be linked in https://github.com/goonstation/goonstation/projects/2 in case anyone wants to fish out removed features later.

**Review by ZeWaka — APPROVED** [review]
code/ci looks correct

**Review by Tarmunora — APPROVED** [review]

**Comment by geneva-breaker** [comment]
> Disabling assjam but leaving the code, from a code perspective, is the worst possible outcome. This would leave a (small) mess of unmaintained code sitting around becoming ever more unable to be turned on the more things change around it. Either keep it enabled or purge the code from master, but don't do it by halves.

unmaintained by whom?

I *was* going to go though and individually purge most of the garbage, it would just be nice to not throw out the whole infrastructure with it, but i guess you could also delete everything right now to save you the pain of a few more commented blocks in our codebase that is already riddled with thousands of lines of commented blocks from mid 2015.

**Review by geneva-breaker — APPROVED** [review]
sure whatever

**Comment by pali6** [comment]
which pieces of the infrastructure would you want to keep?

**Comment by pali6** [comment]
Honestly we could probably make an ass-jam branch with the ass-jam infrastructure and some of the stuff we liked about it and keep it like that until we figure out if we want to use it for some new incarnation of the 13th day of a month.

**Comment by geneva-breaker** [comment]
.github/keylabeler.yml
.github/update_changelog.py
.github/workflows/automaton.yml
.github/workflows/beepsky.yml
.github/workflows/update_changelog.yml
_std/__build.dm
_std/__std.dme
_std/ass_jam.dm
_std/setup.dm
code/client.dm
code/datums/antag_weighting.dm
code/datums/changelog.dm
code/map.dm
code/mob/living/carbon/human/procs/update_icon.dm
code/modules/admin/ass_day.dm
code/world.dm
strings/ass_changelog.txt 

None of these files contain any code that needs to be removed. 
They hold everything required to make any rotating-date or special event content, which is something we do like absolute cavemen, manually for all sorts of stuff. Spooktober? Spacemas? April Fool's? all much better suited using this shit that's been months in the making not just in my stupid code but in the CI support and backend shit. We got Wire to inject the date in 6 different formats so we could later EXPAND the ass jam mechanisms, not wipe them out. 

Nuking this whole setup when you could just revert the actual Ass Jam feature PRs that are causing problems (Timestop, Kinetitech, Ass Day Classic, looking at you!) - and doing it without even talking to me until someone brought the PR to my attention? Rude. 

But seeing as I already **asked** you to just disable it so I could go through and remove the problematic content, and even that's been rejected on the basis of temporarily unused code being bad? More Rude. This is a sledgehammer operation when what it needs is even the slightest bit of discretion. Yeah I could probably put two brain cells together and figure out how to go fetch these parts from the git history but chances are i'll just fuck that up and forget where anything is. 

**Request:**
If you want to do the bare minimum for me (for whenever I get back from this mental disaster) just restore the files listed above, proceed with deleting the rest (i.o.w. 100% of the player-facing content and almost all the IF ELSE blocks that are sooooooooo ugly) and maybe next time you want to erase some coder's whole project, ask them if they mind _before_ making it a public discussion. Enjoy my vacation.

**Comment by pali6** [comment]
I've done my share of trying to improve ass-jam and keep it running, adding both features, posting the *actual forum thread about it*, encouraging people to contribute to it an writing part of the infrastructure around it. Every month I have to defend ass-jam because I'm usually the only person that cares about it and the community comes up with usually valid criticism of the event. I haven't seen you address any of those concerns nor discuss it with the community. If you want to have ass-jam continue please consider listening to feedback.

**Pull request closed without merging**